packages/cli/src/credentials/credentials.service.ee.ts

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable import/no-cycle */
+import { Db } from '..';
+import { CredentialsService } from './credentials.service';
+import { CredentialsEntity } from '../databases/entities/CredentialsEntity';
+import { SharedCredentials } from '../databases/entities/SharedCredentials';
+import { User } from '../databases/entities/User';
+import { RoleService } from '../role/role.service';
+
+export class EECreditentialsService extends CredentialsService {
+	static async isOwned(
+		user: User,
+		credentialId: string,
+	): Promise<{ ownsCredential: boolean; credential?: CredentialsEntity }> {
+		const sharedCredentials = await this.getSharedCredentials(user.id, credentialId, [
+			'credentials',
+		]);
```

Comment:
this again dips into the service vs. namespace problem

Reply:
This time we are inside the class, so I think namespacing is not at play? Mostly aiming to avoid `sharedCredentials` looking like an entity more than a mapping, which is what it is.

Suggestion for the third arg: `{ preload: ['credentials'] }`

Reply:
With this we are moving a bit away from `SharedCredentials` from UM v1, but conceptualizing rows in this table in terms of `sharing` I think is more accurate and descriptive. Having this as a method of the creds sharing service makes sense to me.